presto-main/src/main/java/com/facebook/presto/sql/gen/LambdaBytecodeGenerator.java

```diff
@@ -101,6 +102,19 @@ private LambdaBytecodeGenerator()
         return generateMethodsForLambda(containerClassDefinition, callSiteBinder, cachedInstanceBinder, ImmutableList.of(expression), metadata, sqlFunctionProperties, methodNamePrefix);
     }
 
+    public static Map<LambdaDefinitionExpression, CompiledLambda> generateMethodsForLambda(
+            ClassDefinition containerClassDefinition,
+            CallSiteBinder callSiteBinder,
+            CachedInstanceBinder cachedInstanceBinder,
+            RowExpression expression,
+            Metadata metadata,
+            SqlFunctionProperties sqlFunctionProperties,
+            String methodNamePrefix,
+            Set<LambdaDefinitionExpression> compiledLambdas)
```

Comment:
nit: what about something like `alreadyCompiledLambdas` ?

Reply:
Renamed to `existingCompiledLambdas`. "Compiled" kinda implied "already" so it's a little weird.